packages/backend-common/src/reading/tree/TarArchiveResponse.ts

```diff
@@ -30,6 +30,10 @@ import {
 const TarParseStream = (Parse as unknown) as { new (): ParseStream };
 
 const pipeline = promisify(pipelineCb);
+// Matches a directory name + one `/` at the start of any string,
+// containing any character except `/` one or more times, and ending with a `/`
+// e.g. Will match `dirA/` in `dirA/dirB/file.ext`
+const directoryNameRegex = /^[^\/]+\//;
```

Comment:
Is this backslash on a win system?

Reply:
Hm ok i see the rest of the code uses forward slashes as well

Reply:
Yeah that's why I spared no precaution myself in this PR. Since the Tarball is downloaded from GitHub, kept in memory for all of this, I am assuming `/` separator won't change due to the machine being Windows. But I could totally be wrong.